master/internal/project/authz_basic_impl.go

```diff
@@ -96,6 +96,17 @@ func (a *ProjectAuthZBasic) CanMoveProject(
 	return nil
 }
 
+// CanMoveProjectExperiments returns an error if the user isn't a admin or owner of a project.
+func (a *ProjectAuthZBasic) CanMoveProjectExperiments(
+	curUser model.User, from, to *projectv1.Project,
+) error {
+	if curUser.Admin || from.Immutable || ((model.UserID(from.UserId) == curUser.ID) &&
+		(model.UserID(to.UserId) == curUser.ID)) {
```

Comment:
>user with ownership of both source and destination projects [this doesn't seem right, maybe it should be user owns experiment, and we already have restrictions on viewable source and destination?]

for me it seems that "user is admin or owns experiment" would be a better check for our OSS setup, because it mirrors what we do e.g. for experiment deletion.
maybe something worth running by product team?

Reply:
talked with product; have added a restriction here (webui) and in backend that the experiment can be moved under one of these two conditions:
- admin user
- experiment owner who is moving from the Uncategorized project to their project, or between two of their own projects